MozillaFoundation/foundation.mozilla.org
# added new donate site + landing page model

## Description



Related PRs/issues: #10265

**This PR adds the following:**

- A new folder/app in the codebase titled "donate".
- Factory file that creates a new site titled "Donate" under the domain `donate.localhost`, with a root page of the model `DonateLandingPage`  titled "Donate Now".
- A basic html template for the `DonateLandingPage` model that renders the page title. The front end work is captured in #10266 
- A file with some basic tests for the `DonateLandingPage` model and factory
- An update to `review_app_setup.py` to account for the new donate site
- An update to `fix_local_site_bindings.py to account for the donate site


## Steps to test:

1. Check this branch out and run `inv new-db`
2. Log into the CMS
3. Click "Settings" -> "Sites" and note that there is a new third site titled "Donate".
4. Go back to the CMS homepage and click "Pages"
5. Note that there is a new page titled "Donate Now" alongside "Mozilla Festival" and "Homepage"

## Acceptance Criteria:

- [ ] As a dev, I can run `inv new-db` and find a new site in the CMS titled "Donate". There should be a root level page, next to the foundation homepage and the mozfest homepage, titled "Donate Now". The URL for this page is donate.localhost:8000, and only renders the page's title. 


## Checklist:



**Tests**

**Changes in Models:**

**Documentation:**
- [x] Did I update the READMEs or wagtail documentation? (added a small section for the donate site in the readme, will be updated as I complete more tickets)
